frontend/src/components/common/SearchAndFilters.tsx

```diff
@@ -0,0 +1,246 @@
+import React, { useEffect, useState } from "react";
+import {
+  Accordion,
+  AccordionItem,
+  AccordionButton,
+  AccordionPanel,
+  AccordionIcon,
+  Box,
+  Grid,
+  GridItem,
+  Button,
+  Checkbox,
+  FormControl,
+  FormLabel,
+  Input,
+  InputLeftElement,
+  Icon,
+  InputGroup,
+  Select,
+  Divider,
+  Text,
+} from "@chakra-ui/react";
+import commonAPIClient from "../../APIClients/CommonAPIClient";
+import { LogRecord } from "./types/LogRecord";
+
+type Props = {
+  setLogRecords: React.Dispatch<React.SetStateAction<LogRecord[]>>;
+}
+
+const SearchAndFilters = ({
+  setLogRecords,
+}: Props): React.ReactElement => {
+  /* TODO: change inputs to correct types
+  - this is possible when the data in LogRecords comes from the API since the API will have the employees/attnTos names + userId
+    names + userId and we can query using the /users route based on that
+    e.g.:
+      - for employee -> we'll need to get an employee's ID by their name to pass into the route
+      - for attention to -> same as above
+      - for startDate & endDate the date format should be 'YYYY-MM-DD'
+       - for tags -> change to an actual array of strings instead of a string
+  */
+  // TODO: search by resident
+  const [residents, setResidents] = useState("");
+  const [employees, setEmployees] = useState("");
+  const [startDate, setStartDate] = useState("");
+  const [endDate, setEndDate] = useState("");
+  const [tags, setTags] = useState("");
+  const [attentionTo, setAttentionTo] = useState("");
+  const [building, setBuilding] = useState("");
+  const [flagged, setFlagged] = useState(false);
+
+  useEffect(() => {
+    const getLogRecordsAfterFiltering = async () => {
+      const employeeIds = employees
+        ? employees.replaceAll(`"`, "").split(",")
+        : [];
+      const attentionTos = attentionTo
+        ? attentionTo.replaceAll(`"`, "").split(",")
+        : [];
+      const dateRange = startDate && endDate ? [startDate, endDate] : [];
+
+      const data = await commonAPIClient.filterLogRecords(
+        building,
+        employeeIds,
+        attentionTos,
+        dateRange,
+        tags ? [tags] : [],
+        flagged,
+      );
+      setLogRecords(data);
+    };
+
+    getLogRecordsAfterFiltering();
+  }, [
+    building,
+    employees,
+    attentionTo,
+    startDate,
+    endDate,
+    tags,
+    flagged,
+    setLogRecords,
+  ]);
+
+  // console.log("data", data);
+  return (
+    <Accordion allowToggle padding="20px">
+      <AccordionItem>
+        <AccordionPanel>
+          <Box className="card">
+            <Box className="container" as="span" w="100%">
+              <Text fontSize="12px" fontWeight="700" color="#6D8788">
+                FILTER BY
```

Comment:
In the design, we want to show the first few filter types always, and have the option to show more appear when the open filters button is clicked:
![image](https://github.com/uwblueprint/supportive-housing/assets/57375371/49481f20-7d33-495b-b688-c97712c39277)
![image](https://github.com/uwblueprint/supportive-housing/assets/57375371/39fd631c-6900-4dd6-8084-12bdba22908d)

To see if this is actually doable, I wrote some code and made some progress:
![image](https://github.com/uwblueprint/supportive-housing/assets/57375371/6e2bc47c-8af8-41e3-995e-5e55276ffa17)
![image](https://github.com/uwblueprint/supportive-housing/assets/57375371/41564440-fc8e-4681-945c-92aade9c6df3)

It looks similar to the figma designs, but some things need to be tweaked (like the padding of the first box). I'll share this code with you so you can work off of it.